contract/src/stage2/ft.rs

```diff
@@ -115,7 +111,33 @@ impl DropZone {
         false
     }
 
+    #[private]
+    /// Self callback checks if fungible tokens were successfully refunded. If yes, set keys registered to 0.
+    pub fn ft_resolve_refund(
+        &mut self, 
+        drop_id: DropId,
+        num_to_refund: u64
+    ) -> bool {
+        assert_eq!(env::promise_results_count(), 1, "no promise result");
```

Comment:
what happens when we panic here? we would not have reverted the keys registered so we're in between states?

AFAIK there are 3 promise states and NotReady is never returned. So we must get some promise result right?
- NotReady
- Successful
- Failed

I don't really know what to do, I'm just commenting. We may want to look into this in the future.

Reply:
We would never panic here - it would only ever panic if someone tries to call this fn NOT through a CCC, making the promise result count not equal to 1 and this would never cause an intermediary state because it's not canon.

Reply:
ya ok and it's private so how would they unless it was accidently done through code? Is the assert needed then?

Reply:
Yea maybe it's not... the keys they have can let them sign txns on behalf of our contract but the keys can't call this method so yea I can remove